### PR TITLE
feat: Notes and Status Dropdown

### PR DESCRIPTION
**Pull request by vijaivir**
# Description

This feature adds the ability to collapse/expand the status and notes panel visible on the submissions page and provides a summary in the panel's header. 

I have added an icon next to the 'Status' and 'Notes' heading, expanding the rest of the content if clicked. The Notes panel header displays the total number of notes on the submission. The status panel displays the status and assignee information in the header if the screen is wide enough. 

With the new changes to the status and notes panel layout from the right side to the top of the page, this addition improves readability as the form submission will always be visible regardless of how large the notes are. 

## Collapsed Panels

Screenshot 2024-04-05 at 12 31 41 PM

## Expanded Panels

Screenshot 2024-04-05 at 12 02 29 PM

## Mobile View

Screenshot 2024-04-05 at 11 49 36 AM
Screenshot 2024-04-05 at 12 03 37 PM

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

**Review by jasonchung1871 — APPROVED** [review]

**Comment by WalterMoar** [comment]
Looking good! Just in case you weren't aware, the buttons are laid out a little differently on wide screens (1920x1080 @ 100%). Not sure what the best behaviour here would be, since they're pretty wide buttons already, and making them half a page wide is even bigger!

image

File: app/frontend/src/components/forms/submission/NotesPanel.vue [hunk 1]
```diff
@@ -82,13 +83,19 @@ export default {
...
-    <div class="mt-6 d-flex flex-md-row justify-space-between">
+    <div
+      class="d-flex flex-md-row justify-space-between"
+      @click="showNotesContent = !showNotesContent"
+    >
       <div cols="12" sm="6">
         <h2 class="note-heading" :lang="lang">
           {{ $t('trans.notesPanel.notes') }}
+          <v-icon>{{
+            showNotesContent ? 'mdi:mdi-chevron-down' : 'mdi:mdi-chevron-right'
```
Comment (WalterMoar) [inline app/frontend/src/components/forms/submission/NotesPanel.vue:94]: For LTR languages, would this be better as `mdi:mdi-chevron-left` when it's "closed"?

**Comment by vijaivir** [comment]
> Looking good! Just in case you weren't aware, the buttons are laid out a little differently on wide screens (1920x1080 @ 100%). Not sure what the best behaviour here would be, since they're pretty wide buttons already, and making them half a page wide is even bigger!
> 
> image

I did notice this but it seemed like a separate issue. This is a pretty small PR so I'll see what I can do. Let me consult with Shannon.

File: app/frontend/src/components/forms/submission/NotesPanel.vue (hunk 1 above)
Comment (vijaivir) [reply app/frontend/src/components/forms/submission/NotesPanel.vue:94]: Great suggestion, I didn't even think of that.

**Comment by WalterMoar** [comment]
I'm wondering if people will complain about the extra clicks to see the status and notes. Maybe questions for Shannon, but would it be too cluttered to use the headers to also show the status (and assignee if assigned?) or the number of notes, so that people don't open it if there are zero notes? (Just wondering!)

**Comment by vijaivir** [comment]
> I'm wondering if people will complain about the extra clicks to see the status and notes. Maybe questions for Shannon, but would it be too cluttered to use the headers to also show the status (and assignee if assigned?) or the number of notes, so that people don't open it if there are zero notes? (Just wondering!)

These are good suggestions, I am talking to Shannon about this.

**Comment by vijaivir** [comment]
> Looking good! Just in case you weren't aware, the buttons are laid out a little differently on wide screens (1920x1080 @ 100%). Not sure what the best behaviour here would be, since they're pretty wide buttons already, and making them half a page wide is even bigger!
> 
> image

After consulting with Shannon, I have updated the button sizing for both the Notes and Status panels. The buttons are now set at a fixed width of 200 px.  

Screenshot 2024-04-05 at 11 53 27 AM

**Comment by vijaivir** [comment]
> I'm wondering if people will complain about the extra clicks to see the status and notes. Maybe questions for Shannon, but would it be too cluttered to use the headers to also show the status (and assignee if assigned?) or the number of notes, so that people don't open it if there are zero notes? (Just wondering!)

I have added a summary in the headers: 

Screenshot 2024-04-05 at 12 31 41 PM

**Review by WalterMoar — APPROVED** [review]
If Jason's OK without the tests, then looks good to me! I think the users are going to love this.

**Pull request merged**